evi/event_interface.c

```diff
@@ -533,9 +577,16 @@ static int evi_print_subscriber(mi_item_t *subs_obj, evi_subs_p subs)
 			subs->trans_mod->proto.len, subs->trans_mod->proto.s,
 			socket.len, socket.s) < 0)
 		return -1;
-
 	if (sock->flags & EVI_EXPIRE) {
-		if (add_mi_number(subs_obj, MI_SSTR("expire"), sock->expire) < 0)
+		now = time(0);
+		/* calculate the remaining time for the subscriber to be expired */
+		expiry_countdown = sock->expire - (now - subs->reply_sock->subscription_time);
+		if (expiry_countdown > 0) {
+			if (add_mi_number(subs_obj, MI_SSTR("expire"), expiry_countdown) < 0)
+				return -1;
+		}
```

Comment:
I believe an `else` is required here.
Moreover, to preserve backwards compatibility, I'd suggest to leave the `expire` node as it is, to indicate the expiration time used for the event, and add a new node, let's call it `timeout` or something similar, which contains the number of seconds left.

Reply:
@razvancrainea while adding the `timeout` node does it make sense to add the same node in the area where expire=never

```
	} else {
		if (add_mi_string(subs_obj, MI_SSTR("expire"), MI_SSTR("never")) < 0)
			return -1;
		/* If an event is never expired then its timeout should be never as well*/
		if (add_mi_string(subs_obj, MI_SSTR("timeout"), MI_SSTR("never")) < 0)
			return -1;
	}
```

Reply:
no, definitely not, only when it is still active.

Reply:
I see that `timeout` is still printed for `expire=never` - I don't think it is necessary. And instead of `timeout`, does `remaining` sound better? your call :).